src/main/java/io/fabric8/maven/docker/util/DockerFileUtil.java

```diff
@@ -55,21 +55,38 @@ private DockerFileUtil() {}
      */
     public static List<String> extractBaseImages(File dockerFile, FixedStringSearchInterpolator interpolator) throws IOException {
         List<String[]> fromLines = extractLines(dockerFile, "FROM", interpolator);
+        Map<String, String> args = extractArgs(extractLines(dockerFile, "ARG", interpolator));
         Set<String> result = new LinkedHashSet<>();
         Set<String> fromAlias = new HashSet<>();
         for (String[] fromLine :  fromLines) {
             if (fromLine.length > 1) {
                 if (fromLine.length == 2) { // FROM image:tag use case
-                    result.add(fromLine[1]);
+                    result.add(resolveImageTagFromArgs(fromLine[1], args));
                 } else if (fromLine.length == 4) { // FROM image:tag AS alias use case
                     if (!fromAlias.contains(fromLine[1])) {
-                        result.add(fromLine[1]);
+                        result.add(resolveImageTagFromArgs(fromLine[1], args));
                     }
-                    fromAlias.add(fromLine[3]);
+                    fromAlias.add(resolveImageTagFromArgs(fromLine[3], args));
                 }
             }
         }
-        return result.stream().collect(Collectors.toList());
+        return new ArrayList<>(result);
+    }
+
+    /**
+     * Extract Args from dockerfile. All lines containing ARG is taken.
+     *
+     * @param argLines list of lines containing arguments
+     * @return HashMap of arguments or empty collection if none is found
+     */
+    public static Map<String, String> extractArgs(List<String[]> argLines) {
```

Comment:
Maybe you could inline the usage of `extractLines` ? (e..g change the signature to `extractArgs(String dockerfile, FIxedStringSearchInterpolator interpolator)` and call extract Lines internally)